test/tests/functional/pbs_ralter.py

```diff
@@ -1108,3 +1109,153 @@ def test_conflict_standing_resv_occurrence(self):
+            offset, duration, select="1:ncpus=2", standing=True,
+            rrule="FREQ=DAILY;COUNT=10")
+        self.mom.stop()
+        msg = 'mom is not down'
+        self.assertFalse(self.mom.isUp(), msg)
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
+        self.server.expect(RESV, attrs, id=rid1, max_attempts=30, interval=0.5)
+        self.server.expect(RESV, attrs, id=rid2, max_attempts=30, interval=0.5)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, sequence=1)
+        self.alter_a_reservation(rid2, start2, end2, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, sequence=1)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, interactive=2,
+                                 sequence=2)
+        self.alter_a_reservation(rid2, start2, end2, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, interactive=2,
+                                 sequence=2)
+        self.mom.start()
+
+    def test_sched_cycle_kicks_on_resv_end(self):
+        """
+        This test checks whether the sched cycle gets kicked
+        when the advance reservation ends
+        """
+        duration = 10
+        offset = 5
+
+        rid, _, _ = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=8")
+        attr = {'Resource_List.walltime': '00:00:20'}
+        j = Job(TEST_USER)
+        j.set_sleep_time(10)
+        j.set_attributes(attr)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'},
+                           id=jid, max_attempts=5, interval=0.5)
+        self.scheduler.log_match(
+            jid + ";Job would conflict with reservation or top job",
+            n='ALL',  max_attempts=30, interval=0.5)
+        self.server.log_match(
+            rid.split('.')[0] + ";deleted at request of pbs_server",
+            id=rid.split('.')[0], n='ALL', max_attempts=30, interval=1)
+        self.server.log_match(jid + ";Job Run", id=jid,
+                              n='ALL', max_attempts=30, interval=0.5)
+
+    def test_alter_resv_name(self):
+        """
+        This test checks the alter of reservation name
+        """
+        duration = 30
+        offset = 5
+
+        rid1, _, _ = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=4")
+        rid2, _, _ = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=4", standing=True,
+            rrule="FREQ=DAILY;COUNT=10")
+        attr1 = {ATTR_N: "Adv_Resv"}
+        self.server.alterresv(rid1, attr1)
+        attr2 = {ATTR_N: "Std_Resv"}
+        self.server.alterresv(rid2, attr2)
+        attr1 = {'Reserve_Name': "Adv_Resv"}
+        attr2 = {'Reserve_Name': "Std_Resv"}
```

Comment:
You shouldn't need to redefine attr1 and attr2.

Reply:
I have to as "ATTR_N" resolves to "Job Name" in PTL. Though while setting it takes the "-N" option which works for reservation but same fails for expect in PTL